### PR TITLE
chore: improvement for publish

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

chore/build/publish related change

**Did you add tests for your changes?**

No need

**If relevant, did you update the documentation?**

No need

**Summary**

No :smile: 

**Does this PR introduce a breaking change?**

No need

**Other information**

No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,21 +28,20 @@
...
+    "prepsuite": "node scripts/prepareSuite.js",
     "pretest": "yarn build && yarn lint && yarn prepsuite",
     "test": "jest --reporters=default --reporters=jest-junit",
     "test:cli": "jest test/ --reporters=default --reporters=jest-junit --forceExit",
     "test:packages": "jest packages/ --reporters=default --reporters=jest-junit --forceExit",
     "test:ci": "yarn test:cli && yarn test:packages",
     "test:watch": "jest test/ packages/ --watch",
     "test:smoke": "smoketests/smoketests.sh",
-    "publish:monorepo": "yarn format && yarn test && yarn build && lerna publish -m \"chore: monorepo version update\""
+    "publish:monorepo": "yarn test && yarn build && yarn lerna-changelog && lerna publish -m \"chore: monorepo version update\" --canary --preid rc --dist-tag next"
```
Comment (anshumanv) [inline package.json:44]: canary will append a `SHA` to the package version right?

**Review by anshumanv — APPROVED** [review]
Looks good, one question

File: package.json (hunk 1 above)
Comment (evilebottnawi) [reply package.json:44]: Hope yes :smile: I am still testing and watching everything, so this may not be the last commit

Comment (evilebottnawi) [reply package.json:44]: lerna do not have `--dry` flag to test, so I am afraid of accidentally breaking something

Comment (anshumanv) [reply package.json:44]: Also, does it make sense to run tests here?

We anyways merge on CI green

Comment (anshumanv) [reply package.json:44]: you can try it on your fork first

Comment (anshumanv) [reply package.json:44]: or maybe after npm logout so it won't have access to publish

Comment (evilebottnawi) [reply package.json:44]: > Also, does it make sense to run tests here?

Honestly, I don't have a strong opinion here, some packages do it, some not, but I agree, let's remove, we have CI and should catch problem there, not on publish command

> or maybe after npm logout so it won't have access to publish

that's exactly what i do :smile:

**Review by evenstensberg — APPROVED** [review]

**Review by anshumanv — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Comment by evilebottnawi** [comment]
Flaky tests on node@13 macos, we should use node@14 or increase timeout, maybe it will fix it, let's fix test(s) for watch in future, we focus on release

**Pull request merged**